optaplanner-website-root/content/blog/2023/04/05/BoostVRPListSelectorScalability.adoc

```diff
@@ -0,0 +1,42 @@
+= Boost VRP ListSelector's Scalability with Nearby Selector
+dupliaka
+2023-04-05
+:page-interpolate: true
+:jbake-type: post
+:jbake-tags: insight, design, vehicle routing. move selector
+:jbake-social_media_share_image: nearby-selection.png
+
+As a solver of optimization problems, the VRP ListSelector faces a significant challenge when it comes to scalability.
+
+[quote]
+____
+"`With each additional node in the problem, the complexity of the algorithm increases exponentially, making it difficult to find feasible solutions within a reasonable amount of time.`"
+____
+
+== How to address this challenge?
+
+`NearbyMoveSelector` has been implemented as a tool to reduce the amount of nodes that need to be considered during the optimization process. By pre-calculating a distance matrix and using different distributions to limit the number of moves that need to be considered, the nearbyMoveSelector improves the scalability of the algorithm, making it possible to tackle larger and more complex problems.
+
+But what is the underlying analysis behind the success of the `NearbyMoveSelector`, and where might bottlenecks arise in the optimization process?
+
+At its core, the `NearbyMoveSelector` is a tool for reducing the complexity of the optimization process. By limiting the number of moves that need to be considered, it reduces the number of comparisons required, ultimately leading to a reduction in execution time. This is achieved by pre-calculating a distance matrix, which allows the algorithm to quickly identify the closest nodes and *focus its attention on those that are most relevant*.
+
+== How that improves the calculation?
+
+Let's compare the solver progress over time with `NearbyMoveSelector` and the Default configuration. On the graph below you can see the _optaplanner-benchmarks/.../vehicleRoutingBenchmarkConfigListNearBy.xml_ benchmark results for different types of distributions with *50 Customers* and *10 Vehicles*.  The maximum sampling for the Move Selector in this case is 60 (sum of Planning Values(50) and Planning Entities(10)) that is why the Distribution Size maximum was picked with  https://youexec.com/book-summaries/algorithms-to-live-by[37% of the amount]. By the dynamic of the curve you can see that the PARABOLIC_DESTRIBUTION `NearbyMoveSelector` was the fastest and it overtook `LateAsseptance` being twice as fast.
+
+image::listNearByBenchmark.png[How NearByMoveSelector improves score calculation speed]
+
+== Potential bottlenecks?
+
+However, there are potential bottlenecks that can arise while using of the `NearbyMoveSelector`. For example, if the distribution used to limit the number of moves is too restrictive, it may exclude nodes that could be part of a feasible solution. Similarly, if the amount of supplied moves would be sufficiently large, it can provide a better score while scaling poorly.
```

Comment:
```suggestion
There are potential bottlenecks that can arise while using nearby selection. For example, if the distribution used to limit the number of moves is too restrictive, it may exclude nodes that could be part of a feasible solution. Similarly, if the amount of supplied moves were sufficiently large, it can provide a better score while scaling poorly.
```

Also - "provide a better score while scaling poorly"? This makes no sense to me. Seems like contradicting itself.